lighthouse-core/audits/webfonts.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+const Util = require('../report/v2/renderer/util');
+const WebInspector = require('../lib/web-inspector');
+const UnusedBytes = require('./byte-efficiency/byte-efficiency-audit');
+
+class WebFonts extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'webfonts',
+      description: 'uses font-display',
+      failureDescription: 'Your fonts are blocking FCP!',
+      helpText: 'You should use font-display!!!!',
+      requiredArtifacts: ['traces', 'Fonts'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const trace = artifacts.traces[this.DEFAULT_PASS];
+    const devtoolsLogs = artifacts.devtoolsLogs[this.DEFAULT_PASS];
+    const fontFaces = artifacts.Fonts;
+    const traceOfTabPromise = artifacts.requestTraceOfTab(trace);
+    const networkPromise = artifacts.requestNetworkRecords(devtoolsLogs);
+
+    // Filter font-faces that do not have a display tag with optional or swap
+    const fontsWithoutProperDisplay = fontFaces.filter(fontFace =>
+      !fontFace.display || !['optional', 'swap'].includes(fontFace.display)
+    );
+
+
+    return Promise.all([traceOfTabPromise, networkPromise]).then(([tabTrace, networkRecords]) => {
+      let totalWasted = 0;
+      const fcpInMS = tabTrace.timestamps.firstContentfulPaint / 1000000;
+      const results = networkRecords.filter(record => {
+        const isFont = record._resourceType === WebInspector.resourceTypes.Font;
+        const isLoadedBeforeFCP = record._endTime < fcpInMS || true;
```

Comment:
I thought we settled on flagging all fonts since FCP was too early, is this right @addyosmani @paulirish or am I misremembering?

Reply:
haha my debug is still here with the `|| true` so basically I don't check fcp :p